orca-redis/src/main/java/com/netflix/spinnaker/orca/pipeline/persistence/jedis/RedisExecutionRepository.java

```diff
@@ -510,12 +519,16 @@ Execution retrieveOrchestrationForCorrelationId(
 
   @Override
   public @Nonnull List<Execution> retrieveBufferedExecutions() {
-    // TODO rz - This is definitely not a healthy way to do this.
-    return Observable.concat(
-        retrieve(PIPELINE),
-        retrieve(ORCHESTRATION)
-      )
-      .filter(execution -> execution.getStatus() == ExecutionStatus.BUFFERED)
+    List<Observable<Execution>> observables = allRedisDelegates().stream()
+      .map(d -> Arrays.asList(
+        retrieveObservable(PIPELINE, allBufferedExecutionsKey(PIPELINE), queryAllScheduler, d),
+        retrieveObservable(ORCHESTRATION, allBufferedExecutionsKey(ORCHESTRATION), queryAllScheduler, d)
+      ))
+      .flatMap(List::stream)
+      .collect(Collectors.toList());
+
+    return Observable.merge(observables)
+      .filter(e -> e.getStatus() == BUFFERED)
```

Comment:
I kept the filter here in case this method is called while some other executions get their statuses updated out of band. Maybe not needed.